src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -29,40 +30,89 @@ sealed class TruthTable private (val table: Seq[(BitPat, BitPat)], val default:
+    *
+    * @param table the truth table whose rows will be padded
+    * @return the same truth table but with inputs padded
+    */
+  private def padInputs(table: Iterable[(BitPat, BitPat)]): Iterable[(BitPat, BitPat)] = {
     val inputWidth = table.map(_._1.getWidth).max
-    require(table.map(_._2.getWidth).toSet.size == 1, "output width not equal.")
-    val outputWidth = table.map(_._2.getWidth).head
-    val mergedTable = table.map {
-      // pad input signals if necessary
+    table.map {
       case (in, out) if inputWidth > in.width =>
         (BitPat.N(inputWidth - in.width) ## in, out)
       case (in, out) => (in, out)
     }
-      .groupBy(_._1.toString)
-      .map {
-        case (key, values) =>
-          // merge same input inputs.
-          values.head._1 -> BitPat(s"b${Seq
-            .tabulate(outputWidth) { i =>
-              val outputSet = values
-                .map(_._2)
-                .map(_.rawString)
-                .map(_(i))
-                .toSet
-                .filterNot(_ == '?')
-              require(
-                outputSet.size != 2,
-                s"TruthTable conflict in :\n${values.map { case (i, o) => s"${i.rawString}->${o.rawString}" }.mkString("\n")}"
-              )
-              outputSet.headOption.getOrElse('?')
-            }
-            .mkString}")
-      }
-      .toSeq
+  }
+
+  /** For each duplicated input, collect the outputs into a single Seq.
+    *
+    * @param table the truth table
+    * @return a Seq of tuple of length 2, where the first element is the
+    *         input and the second element is a Seq of OR-ed outputs
+    *         for the input
+    */
+  private def mergeTableOnInputs(table: Iterable[(BitPat, BitPat)]): Seq[(BitPat, Seq[BitPat])] = {
+    groupByIntoSeq(table)(_._1).map {
+      case (input, mappings) =>
+        input -> mappings.map(_._2)
+    }
+  }
+
+  /** Merge two BitPats by OR-ing the values and masks, and setting the
+    *  width to the max width among the two
+    */
+  private def merge(a: BitPat, b: BitPat): BitPat = {
+    new BitPat(a.value | b.value, a.mask | b.mask, a.width.max(b.width))
+  }
+
+  /** Public method for calling with the Espresso decoder format fd
+    *
+    * For Espresso, for each output, a 1 means this product term belongs to the ON-set,
+    *  a 0 means this product term has no meaning for the value of this function".
+    * This is the same as the fd (or f) type in espresso.
+    *
+    * @param table the truth table
+    * @param default the default BitPat is made up of a single bit type, either "?", "0" or "1".
+    *                A default of "?" sets Espresso to fr-format, while a "0" or "1" sets it to the
+    *                fd-format.
+    * @param sort whether to sort the final truth table using BitPat.bitPatOrder
+    * @return a fully built TruthTable
+    */
+  def fromEspressoOutput(table: Iterable[(BitPat, BitPat)], default: BitPat, sort: Boolean = false): TruthTable = {
```

Comment:
Why is this one defaulting to `sort = false`, but regular apply defaults to `sort = true`?